modular_skyrat/modules/implants/code/augments_head.dm

```diff
@@ -0,0 +1,188 @@
+		your heart's rhythm back to normal after its use."
+	icon = 'modular_skyrat/modules/implants/icons/implants.dmi'
+	icon_state = "sandy"
+	slot = ORGAN_SLOT_BRAIN_ANTISTUN
+	zone = BODY_ZONE_HEAD
+	implant_overlay = null
+	implant_color = null
+	actions_types = list(/datum/action/cooldown/sensory_enhancer)
+	w_class = WEIGHT_CLASS_SMALL
+	/// The bodypart overlay datum we should apply to whatever mob we are put into
+	var/datum/bodypart_overlay/simple/sensory_enhancer/bodypart_overlay
+	/// What limb we are inside of, used for tracking when and how to remove our overlays and all that
+	var/obj/item/bodypart/ownerlimb
+
+/obj/item/organ/internal/cyberimp/sensory_enhancer/Insert(mob/living/carbon/receiver, special, drop_if_replaced)
+	var/obj/item/bodypart/limb = receiver.get_bodypart(deprecise_zone(zone))
+
+	. = ..()
+
+	if(!.)
+		return
+
+	if(!limb)
+		return FALSE
+
+	ownerlimb = limb
+	add_to_limb(ownerlimb)
+
+	ADD_TRAIT(receiver, TRAIT_TWITCH_ADAPTED, TRAIT_NARCOTICS)
+
+/obj/item/organ/internal/cyberimp/sensory_enhancer/Remove(mob/living/carbon/organ_owner, special, moving)
+	. = ..()
+
+	if(ownerlimb)
+		remove_from_limb()
+
+	if(organ_owner)
+		organ_owner.update_body_parts()
+
+	REMOVE_TRAIT(organ_owner, TRAIT_TWITCH_ADAPTED, TRAIT_NARCOTICS)
+
+/obj/item/organ/internal/cyberimp/sensory_enhancer/add_to_limb(obj/item/bodypart/bodypart)
+	bodypart_overlay = new()
+	ownerlimb = bodypart
+	ownerlimb.add_bodypart_overlay(bodypart_overlay)
+	owner.update_body_parts()
+	return ..()
+
+/obj/item/organ/internal/cyberimp/sensory_enhancer/remove_from_limb()
+	ownerlimb.remove_bodypart_overlay(bodypart_overlay)
+	QDEL_NULL(bodypart_overlay)
+	ownerlimb = null
+	owner.update_body_parts()
+	return ..()
+
+/obj/item/organ/internal/cyberimp/sensory_enhancer/Destroy()
+	if(ownerlimb)
+		remove_from_limb()
+	return ..()
+
+/datum/bodypart_overlay/simple/sensory_enhancer
+	icon = 'modular_skyrat/modules/implants/icons/implants_onmob.dmi'
+	icon_state = "sandy"
+	layers = EXTERNAL_ADJACENT
+
+/datum/action/cooldown/sensory_enhancer
+	name = "Activate Qani-Laaca System"
+	desc = "Activates your Qani-Laaca computer and grants you its powers. LMB: Short, safer activation. RMB: Longer, more powerful, more dangerous activation."
+	button_icon = 'modular_skyrat/modules/implants/icons/implants.dmi'
+	button_icon_state = "sandy"
+	check_flags = AB_CHECK_CONSCIOUS
+	cooldown_time = 5 MINUTES
+	text_cooldown = TRUE
+	/// Keeps track of how much twitch we inject into people on activation
+	var/injection_amount = 10
+
+/datum/action/cooldown/sensory_enhancer/Trigger(trigger_flags, atom/target)
+	. = ..()
+
+	injection_amount = 10
+
+	if(trigger_flags & TRIGGER_SECONDARY_ACTION)
+		injection_amount = 20
+
+/datum/action/cooldown/sensory_enhancer/Activate(atom/target)
+	. = ..()
+
+	var/mob/living/carbon/human/human_owner = owner
+
+	human_owner.reagents.add_reagent(/datum/reagent/drug/twitch, injection_amount)
```

Comment:
We should probably add a combat log about this, too, it'll make tracking its usage easier.